pages/index.tsx

```diff
@@ -1,7 +1,20 @@
 import type { NextPage } from 'next';
 
+import AuthRequired from '@/components/auth/AuthRequired';
+import Header from '@/components/common/Header';
+import { setLayout } from '@/utils/layout';
+
 const Home: NextPage = () => {
-  return <div>Home</div>;
+  return <AuthRequired>멤버 페이지</AuthRequired>;
 };
 
+setLayout(Home, (page) => {
+  return (
+    <>
+      <Header />
+      {page}
+    </>
+  );
+});
```

Comment:
레아아웃 패턴 좋아요!

그런데 혹시 AuthRequired를 레이아웃에 포함시키는 게 어떨까요?
제 생각엔 인증이 필수라는 정보가 컴포넌트 단에서 필수로 보여지지 않아도 될 것 같은데, 레이아웃으로 빼면 더 심플해지고 좋을 것 같습니당

```suggestion
const Home: NextPage = () => {
  return <Container>멤버 페이지</Container>;
};

setLayout(Home, (page) => {
  return (
    <AuthRequired>
      <Header />
      {page}
    </AuthRequired>
  );
});
```

Reply:
다시 생각해보니 이렇게 하면 '레이아웃'의 기능에 충실하지 못하는 것 같습니다 기존 방식이 더 좋아요 ㅎ